[v5] Angular docs updates (#7942)
AzureAD/microsoft-authentication-library-for-js: lib/msal-angular/docs/redirects.md

# Using redirects in MSAL Angular

When using redirects with MSAL, it is **mandatory** to handle redirects with either the `MsalRedirectComponent` or `handleRedirectObservable`.

Note that specific guidance has been added for using MSAL Angular with Angular standalone components below.

1. [Subscribing to `handleRedirectObservable` manually](#subscribing-to-handleredirectobservable-manually)
1. [`MsalRedirectComponent`](#msalredirectcomponent-a-dedicated-handleredirectobservable-component)
1. [Redirects with standalone components](#redirects-with-standalone-components)

## Subscribing to `handleRedirectObservable` manually

> Many apps using Angular 17 or above will be unable to handle redirects with `MsalRedirectComponent`. If you are unable to bootstrap the `MsalRedirectComponent`, you **must** handle redirects using the `handleRedirectObservable` as follows:

Reviewer: Do we know when an app may or may not be able to use the `MsalRedirectComponent`? Would help to provide more explicit guidance and/or deprecate/remove the `MsalRedirectComponent` altogether if no longer useful.

Author: We do not unfortunately. It doesn't work for us anymore but we're unsure if it won't work for everyone or what factors play into that.

Reviewer: Can we figure it out? 😉

Author: Looks like our modules sample still works with `MsalRedirectComponent` in Angular 20 (not 17 for some reason, still unsure why). Removing this recommendation to subscribe to `handleRedirectObservable` directly.

Reviewer: We can leave the recommendation, let's just clarify when one should use the `MsalRedirectComponent` vs. `handleRedirectObservable`

Author: That clarification would be in the standalone section.

- `handleRedirectObservable()` should be subscribed to on **every** page to which a redirect may occur. Pages protected by the MSAL Guard do not need to subscribe to `handleRedirectObservable()`, as redirects are processed in the Guard.
- Accessing or performing any action related to user accounts should not be done until `handleRedirectObservable()` is complete, as it may not be fully populated until then. Additionally, if interactive APIs are called while `handleRedirectObservables()` is in progress, it will result in an `interaction_in_progress` error. See our document on [events](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-angular/docs/events.md#the-inprogress-observable) for more information on checking for interactions, and our document on [errors](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-angular/docs/errors.md) for details about the `interaction_in_progress` error. 

> - See our [MSAL Angular Modules Sample](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/samples/msal-angular-samples/angular-modules-sample) for examples of this approach.

Reviewer: Is there a specific file in the sample we can point to here?

Author: Changed the link to point to a specific file

- `@azure/msal-angular` provides a dedicated redirect component that can be imported  into your application. You can import the `MsalRedirectComponent` and bootstrap this alongside `AppComponent` in your application on the `app.module.ts`. This will handle all redirects without your components needing to subscribe to `handleRedirectObservable()` manually.
- Pages that wish to perform functions following redirects (e.g. user account functions, UI changes, etc) should subscribe to the `inProgress$` observable, filtering for `InteractionStatus.None`. This will ensure that there are no interactions in progress when performing the functions. Note that the last / most recent `InteractionStatus` will also be available when subscribing to the `inProgress$` observable. Please see our documentation on [events](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-angular/docs/events.md#the-inprogress-observable) for more information on checking for interactions.
- If you do not wish to use the `MsalRedirectComponent`, you **must** handle redirects with `handleRedirectObservable()` yourself, as laid out in the approach above.

> - See our [Angular 15 sample](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/samples/msal-angular-samples/angular-modules-sample/src/app/app.module.ts#L110) for an example of this approach.

Reviewer: Should we remove this link since we won't support Angular 15 anymore?

Author: Since this isn't an Angular 15 sample anymore, the name has been updated

- Pages that wish to perform functions following redirects (e.g. user account functions, UI changes, etc) should subscribe to the `inProgress$` observable, filtering for `InteractionStatus.None`. This will ensure that there are no interactions in progress when performing the functions. Note that the last / most recent `InteractionStatus` will also be available when subscribing to the `inProgress$` observable. Please see our documentation on [events](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-angular/docs/events.md#the-inprogress-observable) for more information on checking for interactions.
- If you do not wish to use the `MsalRedirectComponent`, you **must** handle redirects with `handleRedirectObservable()` yourself, as laid out in the approach above.
- See our [Angular modules sample](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/samples/msal-angular-samples/angular-modules-sample/src/app/app.module.ts#L110) for an example of this approach.

> Note that this approach may not work with Angular standalone components. See the section on [redirects with standalone components below](#redirects-with-standalone-components) for further guidance.

Reviewer: ```suggestion
Note that this approach is not compatible with Angular standalone components. See the section on [redirects with standalone components below](#redirects-with-standalone-components) for further guidance.
```

Reviewer: Also recommend moving this note up to the top of this section and further bringing attention to it using [Alerts](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts)

## Subscribing to `handleRedirectObservable` manually

> Many apps using Angular 17 or above will be unable to handle redirects with `MsalRedirectComponent`. If you are unable to bootstrap the `MsalRedirectComponent`, you **must** handle redirects using the `handleRedirectObservable` as follows:

Reviewer: ```suggestion
Apps using standalone components will be unable to handle redirects with `MsalRedirectComponent`. If you are unable to bootstrap the `MsalRedirectComponent`, you **must** handle redirects using the `handleRedirectObservable` as follows:
```
